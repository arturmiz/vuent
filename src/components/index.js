import './Vnt.vue';
export { default as VntAutosuggest } from './autosuggest';
export { default as VntButton } from './button';
export { default as VntCheckbox } from './checkbox';
export { default as VntDialog } from './dialog';
export { default as VntHeader } from './header';
export { default as VntIcon } from './icon';
export { default as VntInput } from './input';
export { default as VntLink } from './link';
export { default as VntListView } from './listview';
export { default as VntNavView } from './navview';
export { default as VntPivot } from './pivot';
export { default as VntRadio } from './radio';
export { default as VntRating } from './rating';
export { default as VntSelect } from './select';
export { default as VntSlider } from './slider';
export { default as VntToggle } from './toggle';
