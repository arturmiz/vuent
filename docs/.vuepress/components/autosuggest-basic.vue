<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestBasic',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
