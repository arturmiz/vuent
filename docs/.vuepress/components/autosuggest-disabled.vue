<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries" disabled></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestDisabled',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
