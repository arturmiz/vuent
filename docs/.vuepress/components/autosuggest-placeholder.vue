<template>
  <div class="playground">
    <vnt-autosuggest v-model="country"
                     :options="countries"
                     placeholder="Type something..."></vnt-autosuggest>
    <div class="result">country: {{ country }}</div>
  </div>
</template>

<script>
import { VntAutosuggest } from '../../../src/components';

export default {
  name: 'AutosuggestPlaceholder',
  components: {
    VntAutosuggest
  },
  data() {
    return {
      country: null,
      countries: [
        'Australia',
        'Belgium',
        'Canada',
        'China',
        'France',
        'Germany',
        'Poland',
        'United Kingdom',
        'United States',
      ]
    }
  }
}
</script>
