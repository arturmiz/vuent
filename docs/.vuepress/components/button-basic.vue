<template>
  <div class="playground">
    <vnt-button>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonBasic',
  components: {
    VntButton
  }
}
</script>
