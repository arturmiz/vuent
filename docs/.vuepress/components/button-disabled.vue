<template>
  <div class="playground">
    <vnt-button disabled>Button</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonDisabled',
  components: {
    VntButton
  }
}
</script>
