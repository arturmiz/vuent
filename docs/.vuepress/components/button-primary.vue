<template>
  <div class="playground">
    <vnt-button primary>Primary</vnt-button>
  </div>
</template>

<script>
import { VntButton } from '../../../src/components';

export default {
  name: 'ButtonPrimary',
  components: {
    VntButton
  }
}
</script>
