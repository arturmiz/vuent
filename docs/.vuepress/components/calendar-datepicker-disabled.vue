<template>
  <div class="playground">
    <vnt-calendar-datepicker v-model="date" disabled />
    <div class="result">date: {{ date }}</div>
  </div>
</template>

<script>
import { VntCalendarDatepicker } from "../../../src/components";

export default {
  name: "CalendarDatepickerBasic",
  components: {
    VntCalendarDatepicker
  },
  data() {
    return {
      date: "2020-04-05"
    };
  }
};
</script>
