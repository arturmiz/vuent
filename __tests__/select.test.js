import { createLocalVue, mount } from '@vue/test-utils';
import { VntSelect } from '@/components';
import { isInstalled } from './utils';

describe('Select', () => {
  let wrapper;

  test('can be installed separately', () => {
    const localVue = createLocalVue();
    localVue.use(VntSelect);

    expect(isInstalled(localVue, VntSelect)).toBe(true);
  });

  describe('by default', () => {

    beforeAll(() => {
      wrapper = mount(VntSelect);
    });

    test('is not disabled', () => {
      expect(wrapper.vm.disabled).toBe(false);
    });

    test('label text is empty', () => {
      expect(wrapper.vm.label).toBe('');
    });

    test('option\'s label is under "label" property', () => {
      expect(wrapper.vm.optionText).toBe('label');
    });

    test('option\'s value is under "value" property', () => {
      expect(wrapper.vm.optionValue).toBe('value');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can be disabled', () => {

    beforeAll(() => {
      wrapper = mount(VntSelect, {
        propsData: {
          disabled: true,
          options: [
            { value: 1, label: 'List item 1'},
            { value: 2, label: 'List item 2'},
          ]
        }
      });
    });

    test('is disabled', () => {
      expect(wrapper.vm.disabled).toBe(true);
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    describe('then does not allow to show options list', () => {

      beforeAll(() => {
        const control = wrapper.find('.vnt-select__chosen');
        control.trigger('click');
      });

      test('options list is not shown', () => {
        expect(wrapper.vm.isOpened).toBe(false);
      });

      test('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    });

    describe('then does not allow to select a new value', () => {

      beforeAll(() => {
        const control = wrapper.find('.vnt-dropdown-options__item');
        control.trigger('click');
      });

      test('event is not emitted', () => {
        expect(wrapper.emitted()['input']).toBeUndefined();
      });

    });

  });

  describe('allows to select a new value', () => {

    beforeAll(() => {
      wrapper = mount(VntSelect, {
        propsData: {
          options: [
            { value: 1, label: 'List item 1'},
            { value: 2, label: 'List item 2'},
          ]
        }
      });
    });

    describe('shows options list on click', () => {

      beforeAll(() => {
        const control = wrapper.find('.vnt-select__chosen');
        control.trigger('click');
      });

      test('options list is shown', () => {
        expect(wrapper.vm.isOpened).toBe(true);
      });

      test('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
      });

    });

    describe('allows to select a new value', () => {

      beforeAll(() => {
        const option = wrapper.find('.vnt-dropdown-options__item');
        option.trigger('click');
      });

      test('event is emitted', () => {
        expect(wrapper.emitted()['input']).toHaveLength(1);
      });

    });

  });

  describe('can have custom label', () => {

    beforeAll(() => {
      wrapper = mount(VntSelect, {
        propsData: {
          label: 'Custom Select'
        }
      });
    });

    test('is correct', () => {
      expect(wrapper.vm.label).toBe('Custom Select');
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have custom optionText', () => {

    beforeAll(() => {
      wrapper = mount(VntSelect, {
        propsData: {
          optionText: 'text',
          options: [
            { value: 1, text: 'List item 1'},
            { value: 2, text: 'List item 2'},
          ]
        }
      });
    });

    test('renders correctly', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

  });

  describe('can have custom optionValue', () => {

    beforeAll(() => {
      wrapper = mount(VntSelect, {
        propsData: {
          optionValue: 'id',
          options: [
            { id: 123, label: 'List item 1'},
            { id: 256, label: 'List item 2'},
          ]
        }
      });

      const option = wrapper.find('.vnt-dropdown-options__item');
      option.trigger('click');
    });

    test('correct value is resolved', () => {
      const [[newValue]] = wrapper.emitted()['input'];
      expect(newValue).toBe(123);
    });

  });

});
