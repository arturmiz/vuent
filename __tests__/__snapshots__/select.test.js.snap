// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select allows to select a new value shows options list on click renders correctly 1`] = `
<div class="vnt-select">
  <!---->
  <div role="combobox" class="vnt-select__control">
    <div class="vnt-select__chosen"><span class="vnt-select__chosen-label"></span> <span role="presentation" class="vnt-select__options-toggle"></span></div>
    <ul role="listbox" class="vnt-select__options" style="">
      <li role="listitem" class="vnt-select__options-item">List item 1</li>
      <li role="listitem" class="vnt-select__options-item">List item 2</li>
    </ul>
  </div>
</div>
`;

exports[`Select by default renders correctly 1`] = `
<div class="vnt-select">
  <!---->
  <div role="combobox" class="vnt-select__control">
    <div class="vnt-select__chosen"><span class="vnt-select__chosen-label"></span> <span role="presentation" class="vnt-select__options-toggle"></span></div>
    <ul role="listbox" class="vnt-select__options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Select can be disabled renders correctly 1`] = `
<div class="vnt-select">
  <!---->
  <div role="combobox" class="vnt-select__control">
    <div class="vnt-select__chosen vnt-select__chosen--disabled"><span class="vnt-select__chosen-label"></span> <span role="presentation" class="vnt-select__options-toggle"></span></div>
    <ul role="listbox" class="vnt-select__options" style="display: none;">
      <li role="listitem" class="vnt-select__options-item">List item 1</li>
      <li role="listitem" class="vnt-select__options-item">List item 2</li>
    </ul>
  </div>
</div>
`;

exports[`Select can be disabled then does not allow to show options list renders correctly 1`] = `
<div class="vnt-select">
  <!---->
  <div role="combobox" class="vnt-select__control">
    <div class="vnt-select__chosen vnt-select__chosen--disabled"><span class="vnt-select__chosen-label"></span> <span role="presentation" class="vnt-select__options-toggle"></span></div>
    <ul role="listbox" class="vnt-select__options" style="display: none;">
      <li role="listitem" class="vnt-select__options-item">List item 1</li>
      <li role="listitem" class="vnt-select__options-item">List item 2</li>
    </ul>
  </div>
</div>
`;

exports[`Select can have custom label renders correctly 1`] = `
<div class="vnt-select">
  <label class="vnt-select__title">Custom Select</label>
  <div role="combobox" class="vnt-select__control">
    <div class="vnt-select__chosen"><span class="vnt-select__chosen-label"></span> <span role="presentation" class="vnt-select__options-toggle"></span></div>
    <ul role="listbox" class="vnt-select__options" style="display: none;"></ul>
  </div>
</div>
`;

exports[`Select can have custom optionText renders correctly 1`] = `
<div class="vnt-select">
  <!---->
  <div role="combobox" class="vnt-select__control">
    <div class="vnt-select__chosen"><span class="vnt-select__chosen-label"></span> <span role="presentation" class="vnt-select__options-toggle"></span></div>
    <ul role="listbox" class="vnt-select__options" style="display: none;">
      <li role="listitem" class="vnt-select__options-item">List item 1</li>
      <li role="listitem" class="vnt-select__options-item">List item 2</li>
    </ul>
  </div>
</div>
`;
